cmd/query/app/http_handler_test.go

```diff
@@ -162,6 +162,33 @@ func TestGetTraceSuccess(t *testing.T) {
 	assert.Empty(t, response.Errors)
 }
 
+func extractTraces(t *testing.T, response *structuredResponse) []ui.Trace {
+	var traces []ui.Trace
+	b, err := json.Marshal(response.Data)
+	require.NoError(t, err)
+	err = json.Unmarshal(b, &traces)
+	require.NoError(t, err)
+	return traces
+}
+
+func TestGetTraceDedupeSuccess(t *testing.T) {
+	dupedMockTrace := &model.Trace{
+		Spans:    append(mockTrace.Spans, mockTrace.Spans...),
+		Warnings: []string{},
+	}
+
+	ts := initializeTestServer(t)
+	ts.spanReader.On("GetTrace", mock.AnythingOfType("*context.valueCtx"), mock.AnythingOfType("model.TraceID")).
+		Return(dupedMockTrace, nil).Once()
+
+	var response structuredResponse
+	err := getJSON(ts.server.URL+`/api/traces/123456`, &response)
+	require.NoError(t, err)
+	assert.Empty(t, response.Errors)
```

Comment:
Is this the part that's mean to ensure the order? Are you sure they would be in `response.Errors`, and not in `Span.Warnings`?
```suggestion
	assert.Empty(t, response.Errors, "clock skew adjuster logs warnings if spans are not deduped first")
```
Overall though I think it's a weak test, checking for something that's not there (like proving a negative). It would be stronger if it was simultaneously contrasted with the opposite behavior, i.e.
  1. run with only clock skew adjuster, observe the error
  2. run with all adjusters, observe lack of error

Reply:
this check was indeed incorrect, if succeeded with the original adjuster order. I added another check below for span warnings, which was actually showing the errors.

Reply:
thank you very much for the fix, I had started digging on this but wasn't going to have time to look until at least today :)